malariagen_data/anoph/snp_data.py

```diff
@@ -1669,9 +1669,23 @@ def biallelic_snp_calls(
             coords["variant_position"] = ("variants",), variant_position
 
             # Store alleles, transformed.
-            variant_allele_dask = ds_bi["variant_allele"].data
-            variant_allele_out = dask_apply_allele_mapping(
-                variant_allele_dask, allele_mapping, max_allele=1
+
+            variant_allele = ds_bi["variant_allele"].data
+            variant_allele = variant_allele.rechunk((variant_allele.chunks[0], -1))
+
+            # Chunk allele mapping according to same variant_allele.
+            allele_mapping_chunked = da.from_array(
+                allele_mapping, chunks=variant_allele.chunks
+            )
+
+            # Apply allele mapping blockwise to variant_allele.
+            variant_allele_out = da.map_blocks(
+                lambda allele, map: apply_allele_mapping(allele, map, max_allele=1),
+                variant_allele,
+                allele_mapping_chunked,
+                dtype=variant_allele.dtype,
+                chunks=(variant_allele.chunks[0], [2]),
+
```

Comment:
Hi @tristanpwdennis, these changes are no longer necessary as the bug has been fixed in master. The helper function `dask_apply_allele_mapping()` handles the necessary transformation. 

My git-fu is not perfect but there *should* be a simple way to revert these changes so `snp_data.py` matches origin/master. Perhaps something like `git checkout origin/master -- malariagen_data/anoph/snp_data.py` might work?

Reply:
Hi @alimanfoo - sorry - in case you and @jonbrenas can tell I've been struggling a bit here (lack of git-fu, though that website has been helpful). This didn't seem to work as far as I can tell - so any suggestions welcome, thanks! I'll keep looking...

Reply:
Hi @tristanpwdennis, @alimanfoo, I am no expert at git-fu either but I think the easiest solution is to make the change in the shadow PR (where all tests were successful except for the new notebook that uses a local path). As a matter of fact, I did it without any real issue. I would say that, at this point, you have accomplished the job @tristanpwdennis (congrats and thank you, by the way) and all that is left is some light clean-up. @alimanfoo, do you object to considering this PR as a success and closing it (with the option to do more fine-tuning if needed in the shadow PR)?

Reply:
Thanks so much @jonbrenas and @tristanpwdennis 🙏🏻. Having a quick look over this PR there's a couple of small things I noticed, probably worth iterating a bit more here before calling it done. I'll add some comments tomorrow.